test/escape-case.js

```diff
@@ -0,0 +1,61 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/escape-case';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'escape-case',
+	message: 'Use uppercase characters for the value of the escape sequence'
+}];
+
+ruleTester.run('escape-case', rule, {
+	valid: [
+		'const foo = \'\\xA9\';',
+		'const foo = \'\\uD834\';',
+		'const foo = \'\\u{1D306}\';',
+		'const foo = \'\\cA\'',
+		'const foo = `\\xA9`;',
+		'const foo = `\\uD834`;',
+		'const foo = `\\u{1D306}`;',
+		'const foo = `\\cA`;'
+	],
+	invalid: [
+		{
+			code: 'const foo = \'\\xa9\';',
+			errors
+		},
+		{
+			code: 'const foo = \'\\ud834\';',
+			errors
+		},
+		{
+			code: 'const foo = \'\\u{1d306}\';',
+			errors
+		},
+		{
+			code: 'const foo = \'\\ca\';',
+			errors
+		},
+		{
+			code: 'const foo = `\\xa9`;',
+			errors
+		},
+		{
+			code: 'const foo = `\\ud834`;',
```

Comment:
Could you add cases with template literal syntax that contain other things than the element to escape? I'm not sure the fix is well applied (and as @sindresorhus said, the tests should test the fix output).
A few examples:
```js
const foo = `\\ud834foo`;
const foo = `foo\\ud834`;
const foo = `foo \\ud834`;
const foo = `${'\\ud834 foo'} \\ud834`;
// and a few strings containing the same thing
const foo = '\\ud834foo';
const foo = 'foo\\ud834';
const foo = 'foo \\ud834';
```